src/workerd/io/worker-interface.capnp

```diff
@@ -32,12 +32,51 @@ struct Trace @0x8e8d911203762d34 {
     message @2 :Text;
   }
 
+  # Streaming tail workers support an expanded version of Log that supports arbitrary
+  # v8 serialized data or a text message. We define this as a separate new
+  # struct in order to avoid any possible non-backwards compatible disruption to anything
+  # using the existing Log struct in the original trace worker impl. The two structs are
+  # virtually identical with the exception that the message field can be v8 serialized data.
+  struct LogV2 {
+    timestampNs @0 :Int64;
```

Comment:
Seeing as this is streamed and we have a nanosecond granularity on the timestamp are we not afraid this could possibly be used for timing/spectre attacks in some elaborate way?

Reply:
The timestamps will have the same resolution as `Date.now()`

Reply:
`Date.now()` advances whenever there is I/O. It's only constant while executing purely synchronous code.